chore: convert doc references to intra-doc links and document conventions in `AGENTS.md` (#4074)
near/mpc: AGENTS.md

### Arithmetic in Tests
Do not suggest using `checked_add`, `checked_mul`, `checked_sub`, `saturating_add`, or similar checked/saturating arithmetic in test code — this includes `#[cfg(test)]` modules, integration test crates, and e2e test crates. Raw arithmetic operators (`+`, `-`, `*`, `/`) are fine in tests — overflow will cause a panic, which is the desired behavior in tests.

### Trait Naming

> When a trait models a single capability, name it after the action, not as an agent noun derived from it: `ReadContractState`, not `ContractStateReader`. This does not apply to std-idiomatic patterns (`From*`/`Into*`/`To*` conversions, `*Ext` extension traits, `Is*`/`Has*` predicates) or to role nouns for multi-operation abstractions (`SecretsRepository`, `P2PClient`). It applies to new traits and opportunistic renames, not bulk renames of existing traits.

Reviewer: This is an std-idiomatic pattern. Almost all std traits are verbs and not nouns (e.g. `Debug`, `Read`, `Display`, `Copy`, `Clone` etc.). I didn't reflect over `From` and `Into` now, those are prepositions but that's also okay. They still focus on the capability.

Additionally `P2PClient` should be renamed. Probably also `SecretsRepository` although that's a pretty decent name still despite being a noun.

```suggestion
Traits should model a single capability, and be named after the action, not as an agent noun derived from it: `ReadContractState`, not `ContractStateReader`. This follows std-idiomatic patterns (`From*`/`Into*`/`To*` conversions). This applies to new traits and opportunistic renames, existing traits may deviate from this principle.
```

Author: https://github.com/near/mpc/pull/4074/commits/01c84de036939235fdf10ae01b04c5da4171b229 ✅

- explaining common knowledge or terminology;
- burdening the reader with non-relevant information;

> AI-generated code tends to arrive with obvious comments: restating what the next line does, labeling steps (`// setup`, `// send the request`), or narrating the edit that produced the code. Strip these before submitting. Keep a comment only if it says something the code cannot; if a reader can reconstruct it from the names and types on the same screen, delete it. This applies doubly in tests, where the `// Given` / `// When` / `// Then` structure already tells the story.

Reviewer: I felt what we had on keeping comments minimal should be sufficient, but given that we still get these unhelpful comments it's perhaps worth stressing this like you do here.

Author: Yeah, I think it’s worth stressing that comments should be kept to a minimum, since @kevindeforth keeps leaving code review comments about this and it can be automated.

Prefer concise comments, using correct terminology.

> In doc comments, reference other items with rustdoc intra-doc links (`` [`Foo`] ``), not plain `` `Foo` `` backticks. CI rejects broken links in everything rustdoc documents (test code is outside its view), and only linked references are checked at all; a plain backtick reference rots silently when the item is renamed. A backticked word that merely looks like an item (an algorithm name, a type from a crate we do not depend on, a `cfg(test)` item invisible to rustdoc) stays a plain code span.

Reviewer: Yes please, thanks for adding.
